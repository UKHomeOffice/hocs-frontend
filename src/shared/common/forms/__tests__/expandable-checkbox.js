import React from 'react';
import ExpandableCheckbox from '../expandable-checkbox.jsx';

const choice = {
    label: '__label__',
    value: '__value__'
};

describe('When the component is rendered', () => {
    it('should match the snapshot', () => {
        expect(render(<ExpandableCheckbox choice={choice} data={{}} name="__name__" updateState={() => { }} />))
            .toMatchSnapshot();
    });
});

describe('When the component is rendered and it has child components', () => {
    const items = [{ component: 'inset', props: {} }, { component: 'inset', props: {} }, { component: 'inset', props: {} }];

    it('should match the snapshot - the child components should not be initially displayed by default', () => {
        expect(render(<ExpandableCheckbox choice={choice} data={{}} items={items} name="__name__" updateState={() => { }} />))
            .toMatchSnapshot();
    });
});

describe('When the component is rendered and it has child components and the initiallyOpen prop', () => {
    const items = [{ component: 'inset', props: {} }, { component: 'inset', props: {} }, { component: 'inset', props: {} }];

    it('should match the snapshot - the child components should be initially displayed', () => {
        expect(render(<ExpandableCheckbox choice={choice} data={{}} initiallyOpen={true} items={items} name="__name__" updateState={() => { }} />))
            .toMatchSnapshot();
    });
});

describe('When the component is rendered', () => {
    it('should render checked if the value matches', () => {
        expect(render(<ExpandableCheckbox choice={choice} data={{}} name="__name__" updateState={() => { }} value="__value__" />))
            .toMatchSnapshot();
    });
});

describe('When the component is rendered', () => {
    it('should render unchecked if the value doesnt match', () => {
        expect(render(<ExpandableCheckbox choice={choice} data={{}} name="__name__" updateState={() => { }} value="__notvalue__" />))
            .toMatchSnapshot();
    });
});
