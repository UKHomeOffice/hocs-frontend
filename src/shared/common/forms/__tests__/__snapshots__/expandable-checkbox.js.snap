// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When the component is rendered and it has child components and the initiallyOpen prop should match the snapshot - the child components should be initially displayed 1`] = `
<div
  class="govuk-grid-row selectable-details"
>
  <div
    class="selectable-details-header"
  >
    <div
      class="govuk-checkboxes"
      style="display:inline-flex"
    >
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="details-checkbox-__name__"
          name="details-checkbox-__name__"
          type="checkbox"
          value="__choiceValue__"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="details-checkbox-__name__"
        >
          __label__
        </label>
      </div>
    </div>
    <div
      class="selectable-details-link"
    >
      <span>
        Details
      </span>
    </div>
  </div>
  <div
    class="selectable-details-content"
  >
    <p
      class="govuk-inset-text"
    >
       
    </p>
    <p
      class="govuk-inset-text"
    >
       
    </p>
    <p
      class="govuk-inset-text"
    >
       
    </p>
  </div>
</div>
`;

exports[`When the component is rendered and it has child components should match the snapshot - the child components should not be initially displayed by default 1`] = `
<div
  class="govuk-grid-row selectable-details"
>
  <div
    class="selectable-details-header"
  >
    <div
      class="govuk-checkboxes"
      style="display:inline-flex"
    >
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="details-checkbox-__name__"
          name="details-checkbox-__name__"
          type="checkbox"
          value="__choiceValue__"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="details-checkbox-__name__"
        >
          __label__
        </label>
      </div>
    </div>
    <div
      class="selectable-details-link"
    >
      <span>
        Details
      </span>
    </div>
  </div>
</div>
`;

exports[`When the component is rendered should match the snapshot 1`] = `
<div
  class="govuk-grid-row selectable-details"
>
  <div
    class="selectable-details-header"
  >
    <div
      class="govuk-checkboxes"
      style="display:inline-flex"
    >
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="details-checkbox-__name__"
          name="details-checkbox-__name__"
          type="checkbox"
          value="__choiceValue__"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="details-checkbox-__name__"
        >
          __label__
        </label>
      </div>
    </div>
    <div
      class="selectable-details-link"
    >
      <span>
        Details
      </span>
    </div>
  </div>
</div>
`;

exports[`When the component is rendered should render checked if the value matches 1`] = `
<div
  class="govuk-grid-row selectable-details"
>
  <div
    class="selectable-details-header"
  >
    <div
      class="govuk-checkboxes"
      style="display:inline-flex"
    >
      <div
        class="govuk-checkboxes__item"
      >
        <input
          checked=""
          class="govuk-checkboxes__input"
          id="details-checkbox-__name__"
          name="details-checkbox-__name__"
          type="checkbox"
          value="__value__"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="details-checkbox-__name__"
        >
          __label__
        </label>
      </div>
    </div>
    <div
      class="selectable-details-link"
    >
      <span>
        Details
      </span>
    </div>
  </div>
</div>
`;

exports[`When the component is rendered should render unchecked if the value doesnt match 1`] = `
<div
  class="govuk-grid-row selectable-details"
>
  <div
    class="selectable-details-header"
  >
    <div
      class="govuk-checkboxes"
      style="display:inline-flex"
    >
      <div
        class="govuk-checkboxes__item"
      >
        <input
          class="govuk-checkboxes__input"
          id="details-checkbox-__name__"
          name="details-checkbox-__name__"
          type="checkbox"
          value="__value__"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="details-checkbox-__name__"
        >
          __label__
        </label>
      </div>
    </div>
    <div
      class="selectable-details-link"
    >
      <span>
        Details
      </span>
    </div>
  </div>
</div>
`;
